starknet_py/net/client_models.py

```diff
@@ -192,7 +195,11 @@ class TransactionReceipt:
 
     events: List[Event] = field(default_factory=list)
     l2_to_l1_messages: List[L2toL1Message] = field(default_factory=list)
+
+    # gateway only
     l1_to_l2_consumed_message: Optional[L1toL2Message] = None
+    execution_resources: Optional[dict] = None
+    transaction_index: Optional[int] = None
```

Comment:
Do we need both `GatewayBlockTransactionReceipt` and this?

Reply:
We could, however more fields in `TransactionReceipt` would have to be marked as Optional